src/EditorFeatures/Test/AbstractCommandHandlerTestState.cs

```diff
@@ -64,25 +73,27 @@ internal abstract class AbstractCommandHandlerTestState : IDisposable
 
             if (cursorDocument.AnnotatedSpans.TryGetValue("Selection", out selectionSpanList))
             {
-                var span = selectionSpanList.First();
+                var firstSpan = selectionSpanList.First();
+                var lastSpan = selectionSpanList.Last();
                 var cursorPosition = cursorDocument.CursorPosition.Value;
 
-                Assert.True(cursorPosition == span.Start || cursorPosition == span.Start + span.Length,
+                Assert.True(cursorPosition == firstSpan.Start || cursorPosition == lastSpan.End,
                     "cursorPosition wasn't at an endpoint of the 'Selection' annotated span");
 
                 _textView.Selection.Select(
-                    new SnapshotSpan(_subjectBuffer.CurrentSnapshot, new Span(span.Start, span.Length)),
-                    isReversed: cursorPosition == span.Start);
+                    new SnapshotSpan(_subjectBuffer.CurrentSnapshot, new Span(firstSpan.Start, firstSpan.Length)),
+                    isReversed: cursorPosition == firstSpan.Start);
 
                 if (selectionSpanList.Count > 1)
                 {
                     _textView.Selection.Mode = TextSelectionMode.Box;
-                    foreach (var additionalSpan in selectionSpanList.Skip(1))
-                    {
-                        _textView.Selection.Select(
-                            new SnapshotSpan(_subjectBuffer.CurrentSnapshot, new Span(additionalSpan.Start, additionalSpan.Length)),
-                            isReversed: false);
-                    }
+                    SnapshotPoint boxSelectionStart = new SnapshotPoint(_subjectBuffer.CurrentSnapshot, firstSpan.Start);
+                    SnapshotPoint boxSelectionEnd = new SnapshotPoint(_subjectBuffer.CurrentSnapshot, lastSpan.End);
+                    bool isReversed = cursorPosition == firstSpan.Start;
```

Comment:
Extract this local and use it above?  Also, will this do the right thing when the span is empty?  (Or is that impossible here?)